src/libsyntax/parse/parser.rs

```diff
@@ -1205,6 +1205,23 @@ impl<'a> Parser<'a> {
         Ok(())
     }
 
+    /// Parses an inherited visibility (`VisibilityKind::Inherited`)
+    /// or errors with parser recovery. See #65041 for more context.
+    fn parse_visibility_inherited(&mut self) -> PResult<'a, ()> {
+        let snapshot_expected = self.expected_tokens.clone();
```

Comment:
Maybe we should instead do:
```suggestion
        let snapshot_expected = mem::take(&mut self.expected_tokens);
```
which will likely allocate again, but probably less work than `.clone()` overall.

Reply:
This is moot now.